build.gradle.kts

```diff
@@ -1,40 +1,23 @@
 import org.gradle.api.publish.PublishingExtension
 import org.gradle.api.publish.maven.MavenPublication
 
-buildscript {
-    var kotlin_version: String by extra
-    kotlin_version = "1.1.1"
-
-    repositories {
-        maven { setUrl("http://dl.bintray.com/kotlin/kotlin-eap-1.1") }
-        maven { setUrl("https://plugins.gradle.org/m2/") }
-        mavenCentral()
-    }
-    dependencies {
-        classpath("org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version")
-        classpath("com.gradle.publish:plugin-publish-plugin:0.9.7")
-    }
+plugins {
+    id("org.jetbrains.kotlin.jvm").version("1.1.1")
+    id("com.gradle.plugin-publish").version("0.9.7")
+    id("maven-publish")
 }
 
-apply {
-    plugin("kotlin")
-    plugin("maven-publish")
-    plugin("com.gradle.plugin-publish")
-}
 group = "org.jlleitschuh.gradle"
 version = "1.0.2"
 
-val kotlin_version: String by extra
-
 repositories {
-    maven { setUrl("http://dl.bintray.com/kotlin/kotlin-eap-1.1") }
     mavenCentral()
 }
 
 dependencies {
-    compile(gradleApi())
-    compile(gradleScriptKotlinApi())
-    compile(kotlinModule("gradle-plugin"))
+    compileOnly(gradleApi())
+    compileOnly(gradleScriptKotlinApi())
+    compileOnly(kotlinModule("gradle-plugin"))
```

Comment:
I believe the user of the plugin will need to have a version of gradle which supports build script kotlin and also have the JetBrains kotlin plugin in order to use your plugin anyway. Keeping your specific dependency from becoming a transitive dependency is probably more flexible.

I'm not an expert on publishing gradle plugins, but it appeared what was happening was you had an older version of gradle wrapper and since it was used to build the published plugin it brought an older version of kotlin with it as a transitive dependency.

I can do some additional testing with other user project configurations to make sure.